File: race-administrator/src/main/java/se/cag/labs/raceadmin/UserResult.java
package se.cag.labs.raceadmin;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@Builder
@AllArgsConstructor
public class UserResult {
  private long created = System.currentTimeMillis();
  private User user;
  private long time;
  private long splitTime;
  private ResultType result;

  public enum ResultType {
    FINISHED, WALKOVER, DISQUALIFIED
  }
}

File: admin-frontend/src/app/shared/backend/UserResult.ts
import {User} from "./User";

export type ResultType = 'FINISHED'|'WALKOVER'|'DISQUALIFIED';

export class UserResult {
  readonly id: string;
  readonly created: number;
  readonly user: User;
  readonly time: number;
  readonly splitTime: number;
  readonly result: ResultType;
}
